src/test/lib/storage/storage_manager_test.cpp

```diff
@@ -236,20 +236,102 @@ TEST_F(StorageManagerTest, HasPreparedPlan) {
   EXPECT_EQ(sm.has_prepared_plan("first_prepared_plan"), true);
 }
 
-TEST_F(StorageManagerTest, WriteMaxNumberOfChunksToFileMid) {
-  const auto file_name = "test_chunks_file.bin";
+TEST_F(StorageManagerTest, PersistencyOffsetEnds) {
+  auto& sm = Hyrise::get().storage_manager;
+
+  const auto COLUMN_COUNT = uint32_t{5};
+  const auto ROW_COUNT = uint32_t{500};
+
+  const auto segment_offsets = sm.generate_segment_offset_ends(
+    StorageManagerTestUtil::create_dictionary_segment_chunk(ROW_COUNT, COLUMN_COUNT)
+  );
+
+  const auto expected_offsets = std::vector<uint32_t>{ 3036, 4548, 5728, 6740, 7652 };
+
+  EXPECT_EQ(expected_offsets.size(), segment_offsets.size());
+  EXPECT_EQ(expected_offsets, segment_offsets);
+}
+
+TEST_F(StorageManagerTest, PersistencyDifferentChunks) {
+  auto& sm = Hyrise::get().storage_manager;
+
+  const auto file_name = "test_various_chunks.bin";
+  const auto MAX_CHUNK_COUNT = sm.get_max_chunk_count_per_file();
+  
   std::remove(file_name);
+
+  const auto small_chunk1 = StorageManagerTestUtil::create_dictionary_segment_chunk(100, 5);
+  const auto small_chunk2 = StorageManagerTestUtil::create_dictionary_segment_chunk(123, 1);
+  const auto small_chunk3 = StorageManagerTestUtil::create_dictionary_segment_chunk(40, 10);
+  const auto small_chunk4 = StorageManagerTestUtil::create_dictionary_segment_chunk(5, 10);
+  const auto small_chunk5 = StorageManagerTestUtil::create_dictionary_segment_chunk(400, 30);
+  const auto small_chunk6 = StorageManagerTestUtil::create_dictionary_segment_chunk(77, 3);
+  const auto small_chunk7 = StorageManagerTestUtil::create_dictionary_segment_chunk(20, 9);
+
+  const auto medium_chunk1 = StorageManagerTestUtil::create_dictionary_segment_chunk(3'000, 50);
+  const auto medium_chunk2 = StorageManagerTestUtil::create_dictionary_segment_chunk(4'500, 70);
+  const auto medium_chunk3 = StorageManagerTestUtil::create_dictionary_segment_chunk(2'000, 25);
+  const auto medium_chunk4 = StorageManagerTestUtil::create_dictionary_segment_chunk(1'500, 55);
+  const auto medium_chunk5 = StorageManagerTestUtil::create_dictionary_segment_chunk(4'500, 20);
+
+  const auto big_chunk1 = StorageManagerTestUtil::create_dictionary_segment_chunk(50'000, 500);
+  const auto big_chunk2 = StorageManagerTestUtil::create_dictionary_segment_chunk(100'000, 30);
+  const auto big_chunk3 = StorageManagerTestUtil::create_dictionary_segment_chunk(90'000, 400);
+  const auto big_chunk4 = StorageManagerTestUtil::create_dictionary_segment_chunk(80'000, 200);
```

Comment:
thats a lot of different chunks 😁 I think 5 could have been enough :D

Reply:
Maybe take
const auto small_chunk1 = StorageManagerTestUtil::create_dictionary_segment_chunk(1, 1);
  const auto small_chunk2 = StorageManagerTestUtil::create_dictionary_segment_chunk(100, 1);
  const auto small_chunk3 = StorageManagerTestUtil::create_dictionary_segment_chunk(1, 100);
  const auto small_chunk4 = StorageManagerTestUtil::create_dictionary_segment_chunk(100, 100);

const auto big_chunk1 = StorageManagerTestUtil::create_dictionary_segment_chunk(50'000, 1);
const auto big_chunk2 = StorageManagerTestUtil::create_dictionary_segment_chunk(1, 500);
const auto big_chunk3 = StorageManagerTestUtil::create_dictionary_segment_chunk(90'000, 400);
const auto big_chunk4 = StorageManagerTestUtil::create_dictionary_segment_chunk(80'000, 200);

Ok, no wI see why you chose so many chunks :D but maybe a more systematic approach regarding the variation of th erow/column counts is useful. Bu topen to discuss here :)

Reply:
Good point. Is implemented 👍🏼